### PR TITLE
doc: add LUKS examples & operator notes

### LINKED ISSUE #1020: LUKS: examples & operator notes
# Feature Request #

## Desired Feature ##

Both examples & operator notes should be added for configuring LUKS based devices.

### PR DESCRIPTION
**Pull request by arithx**
Closes #1020

### CODE REVIEW

File: doc/examples.md [hunk 1]
```diff
@@ -240,5 +240,47 @@ Users can be added to an OS with the `passwd.users` key which takes a list of ob
...
 
+## Create a LUKS Volume
+
+This config will set up a key-file based LUKS2 volume, `data`, put a filesystem on the volume, and write a mount unit (shown below) to automatically mount the volume to `/var/lib/data`.
+
+```json ignition
+{
+  "ignition": {"version": "3.2.0-experimental"},
+  "storage": {
+    "filesystems": [{
```
Comment (jlebon) [inline doc/examples.md:251]: Would read better I think if we put `filesystems` after `luks`.

File: doc/examples.md [hunk 2]
```diff
@@ -240,5 +240,47 @@ Users can be added to an OS with the `passwd.users` key which takes a list of ob
...
+      "path": "/var/lib/data",
+      "device": "/dev/disk/by-id/dm-name-data",
```
Comment (jlebon) [inline doc/examples.md:253]: `/dev/mapper/data`?

File: doc/examples.md [hunk 3]
```diff
@@ -240,5 +240,47 @@ Users can be added to an OS with the `passwd.users` key which takes a list of ob
...
+    "luks": [{
+      "name": "data",
+      "device": "/dev/sdb"
+    }]
+  },
+  "systemd": {
+    "units": [{
+      "name": "var-lib-data.mount",
+      "enabled": true,
+      "contents": "[Mount]\nWhat=/dev/disk/by-id/dm-name-data\nWhere=/var/lib/data\nType=ext4\n\n[Install]\nWantedBy=local-fs.target"
```
Comment (jlebon) [inline doc/examples.md:266]: `What=/dev/disk/by-label/DATA`? (And below.)

File: doc/examples.md (hunk 2 above)
Comment (arithx) [reply doc/examples.md:253]: Either work, I went with the persistent block device name to fit in line with the rest of the docs.

**Comment by arithx** [comment]
Updated

**Pull request merged**